mavros/src/plugins/home_position.cpp

```diff
@@ -0,0 +1,140 @@
+/**
+ * @brief HomePosition plugin
+ * @file home_position.cpp
+ * @author Thomas Stastny <thomas.stastny@mavt.ethz.ch>
+ * @author Nuno Marques <n.marques21@hotmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2017 Thomas Stastny, Nuno Marques.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+#include <mavros/mavros_plugin.h>
+#include <eigen_conversions/eigen_msg.h>
+
+#include <mavros_msgs/HomePosition.h>
+
+namespace mavros {
+namespace std_plugins {
+/**
+ * @brief home position plugin.
+ *
+ * Publishes home position.
+ */
+class HomePositionPlugin : public plugin::PluginBase {
+public:
+	HomePositionPlugin() :
+		hp_nh("~")
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		home_position_pub = hp_nh.advertise<mavros_msgs::HomePosition>("get",10);
```

Comment:
https://github.com/mavlink/mavros/blob/master/mavros/src/plugins/waypoint.cpp#L151 did you see that `true`?
http://docs.ros.org/kinetic/api/roscpp/html/classros_1_1NodeHandle.html#a6b655c04f32c4c967d49799ff9312ac6

So yes, it is not storing, it is latching publisher.